qemu/tests/sgx_basic.py

```diff
@@ -0,0 +1,44 @@
+from virttest import error_context
+from virttest.utils_misc import verify_dmesg
+
+from provider.sgx import SGXHostCapability
+from provider.sgx import SGXChecker
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu sgx basic test:
+    1. Check host sgx capability
+    2. Boot sgx VM
+    3. Check sgx qmp cmd
+    4. Verify sgx flags and epc size
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Start sgx test", test.log.info)
+    timeout = float(params.get("login_timeout", 240))
+
+    sgx_cap = SGXHostCapability(test, params)
+    sgx_cap.validate_sgx_cap()
+    host_total_epc_size = sgx_cap.host_epc_size
+    if params.get("monitor_expect_nodes"):
+        sgx_cap.validate_numa_node_count()
+
+    vm = env.get_vm(params["main_vm"])
+    vm.create()
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    verify_dmesg()
+    dmesg_output = session.cmd_output(params["guest_sgx_check"],
+                                      timeout=240).strip()
+    session.close()
+
+    test_check = SGXChecker(test, params, vm)
```

Comment:
hi @zixi-chen , your code is OK for me here, but I just make some comments for your reference in future case-dev.

For the functional verification part, we need to be careful if the code 'changes' anything, e.g. dd a new file, then we need to restore it after testing, e.g. remove the file
try:
checking xxx (dd of=/path/to/file)
...
finally:
restoring env (rm /path/to/file)

Look at your code here, if any verify* function fails, then vm.destroy will never be called, but it's OK for we set kill_vm=yes in cfg, which means that let VT take care of it. Just take care of this kinds of things in future :-D

Reply:
Thanks Zhenchao, I'll pay attention to things like this.